packages/cli/src/commands/dev.ts

```diff
@@ -0,0 +1,165 @@
+import { Command, flags } from '@oclif/command'
+import { readdirSync, existsSync, readFileSync } from 'fs'
+import { join } from 'path'
+import LCD from '@mesg/api/lib/lcd'
+import API from '@mesg/api'
+import Listr from 'listr'
+import dotenv from 'dotenv'
+import * as Environment from '../utils/environment-tasks'
+import * as Service from '../utils/service'
+import * as Runner from '../utils/runner'
+import * as Process from '../utils/process'
+import * as base58 from '@mesg/api/lib/util/base58'
+import version from '../version'
+import { IService } from '@mesg/api/lib/service-lcd'
+import { IExecution } from "@mesg/api/lib/execution";
+import { Stream as GRPCStream } from "@mesg/api/lib/util/grpc";
+import { ExecutionStatus } from '@mesg/api/lib/types'
+import { IProcess } from '@mesg/api/lib/process-lcd'
+import { IRunner } from '@mesg/api/lib/runner-lcd'
+import chalk from 'chalk'
+import { decode } from '@mesg/api/lib/util/encoder'
+
+const ipfsClient = require('ipfs-http-client')
+
+export default class Dev extends Command {
+  static description = 'Start a dev environment for your project'
+
+  static flags = {
+    version: flags.string({ name: 'Engine version', default: version.engine }),
+    pull: flags.boolean({ name: 'Force to pull the docker image', default: false }),
+  }
+
+  static args = [{
+    name: 'PATH',
+    description: 'Path of your project',
+    default: './'
+  }]
+
+  private lcdEndpoint = 'http://localhost:1317'
+  private lcd = new LCD(this.lcdEndpoint)
+  private ipfsClient = ipfsClient('ipfs.app.mesg.com', '5001', { protocol: 'http' })
+  private grpc = new API('localhost:50052')
+  private logs: GRPCStream<IExecution>
+  private services: IService[] = []
+  private processes: IProcess[] = []
+  private runners: IRunner[] = []
+
+  async run() {
+    const { args, flags } = this.parse(Dev)
+
+    if (!existsSync(join(args.PATH, 'services'))) this.error(`${args.PATH} is not a recognized as a MESG project`)
```

Comment:
why forcing to have a services folder?
what about application that use only services from github?

Reply:
initially, the idea was to have the `link` command that will populate the `service` directory with the one from github or any to have everything in local.
The goal is also to always have the same structure even if it's empty (can put a `.gitkeep`) just to keep it as a framework where you don't necessarily use all the features